File: app/src/main/java/de/xikolo/controllers/dialogs/UnenrollDialog.java
package de.xikolo.controllers.dialogs;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;

import de.xikolo.R;
import de.xikolo.controllers.dialogs.base.BaseDialogFragment;

public class UnenrollDialog extends BaseDialogFragment {

    public static final String TAG = UnenrollDialog.class.getSimpleName();

    private UnenrollDialogListener listener;

    public void setUnenrollDialogListener(UnenrollDialogListener listener) {
        this.listener = listener;
    }

    @Override
    @NonNull
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity(), R.style.AppTheme_Dialog);
        builder.setMessage(R.string.dialog_unenroll_message)
                .setTitle(R.string.dialog_unenroll_title)
                .setPositiveButton(R.string.dialog_unenroll_yes, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        if (listener != null) {
                            listener.onDialogPositiveClick(UnenrollDialog.this);
                        }
                    }
                })
                .setNegativeButton(R.string.dialog_negative, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        UnenrollDialog.this.getDialog().cancel();
                    }
                })
                .setCancelable(true);

        AlertDialog dialog = builder.create();
        dialog.setCanceledOnTouchOutside(true);

        return dialog;
    }

    public interface UnenrollDialogListener {
        void onDialogPositiveClick(DialogFragment dialog);
    }

}

File: app/src/main/java/de/xikolo/controllers/dialogs/UnenrollDialog.kt
package de.xikolo.controllers.dialogs

import android.app.Dialog
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import de.xikolo.R
import de.xikolo.controllers.dialogs.base.BaseDialogFragment

class UnenrollDialog : BaseDialogFragment() {

    companion object {
        @JvmField val TAG: String = UnenrollDialog::class.java.simpleName
    }

    var listener: Listener? = null

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(activity!!, R.style.AppTheme_Dialog)
        builder.setMessage(R.string.dialog_unenroll_message)
            .setTitle(R.string.dialog_unenroll_title)
            .setPositiveButton(R.string.dialog_unenroll_yes) { _, _ -> listener?.onDialogPositiveClick(this) }
            .setNegativeButton(R.string.dialog_negative) { _, _ -> dialog?.cancel() }
            .setCancelable(true)

        val dialog = builder.create()
        dialog.setCanceledOnTouchOutside(true)

        return dialog
    }

    interface Listener {
        fun onDialogPositiveClick(dialog: DialogFragment)
    }

}
